fda/functional_data.py

```diff
@@ -0,0 +1,512 @@
+        functional samples at the observation points.
+
+        Returns:
+            numpy.darray: Matrix of covariances.
+
+        """
+        pass
+
+    @abstractmethod
+    def round(self, decimals=0):
+        """Evenly round to the given number of decimals.
+
+        Args:
+            decimals (int, optional): Number of decimal places to round to.
+                If decimals is negative, it specifies the number of
+                positions to the left of the decimal point. Defaults to 0.
+
+        Returns:
+            :obj:FData: Returns a FData object where all elements
+            in its data_matrix are rounded .The real and
+            imaginary parts of complex numbers are rounded separately.
+
+        """
+        pass
+
+    @abstractmethod
+    def to_grid(self, eval_points=None):
+        """Return the discrete representation of the object.
+
+        Args:
+            eval_points (array_like, optional): Set of points where the
+                functions are evaluated.
+
+        Returns:
+              FDataGrid: Discrete representation of the functional data
+              object.
+        """
+
+        pass
+
+    @abstractmethod
+    def to_basis(self, basis, eval_points=None, **kwargs):
+        """Return the basis representation of the object.
+
+        Args:
+            basis(Basis): basis object in which the functional data are
+                going to be represented.
+            **kwargs: keyword arguments to be passed to
+                FDataBasis.from_data().
+
+        Returns:
+            FDataBasis: Basis representation of the funtional data
+            object.
+        """
+
+        pass
+
+    @abstractmethod
+    def concatenate(self, other):
+        """Join samples from a similar FData object.
+
+        Joins samples from another FData object if it has the same
+        dimensions and has compatible representations.
+
+        Args:
+            other (:class:`FData`): another FData object.
+
+        Returns:
+            :class:`FData`: FData object with the samples from the two
+            original objects.
+        """
+
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        """Return repr(self)."""
+
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        """Return str(self)."""
+
+        pass
+
+    @abstractmethod
+    def __getitem__(self, key):
+        """Return self[key]."""
+
+        pass
+
+    @abstractmethod
+    def __add__(self, other):
+        """Addition for FData object."""
+
+        pass
+
+    @abstractmethod
+    def __radd__(self, other):
+        """Addition for FData object."""
+
+        pass
+
+    @abstractmethod
+    def __sub__(self, other):
+        """Subtraction for FData object."""
+
+        pass
+
+    @abstractmethod
+    def __rsub__(self, other):
+        """Right subtraction for FData object."""
+
+        pass
+
+    @abstractmethod
+    def __mul__(self, other):
```

Comment:
Cases like multiplication, addition and division by a number could be done exactly, I think that this cases should be called using this operators. 
Operations between FDataBasis (or FData) could be done approximately with their own function, although I am in favour of using the operators for such cases as well.

Reply:
You are right. In the scalar multiplication case it can be useful to have those operators, and they are exact.

I am not sure about the approximations using the operators, though. In fact, exact multiplication IS possible in general, AFAIK, if we allow heterogeneus basis functions. I still think the operators should be reserved for exact operations.

Reply:
So, shall I delete the methods from FData and FDataBasis for the moment?